### PR TITLE
upgrade coyote

### PR DESCRIPTION
**Pull request by pdeligia**
@ankushdesai this PR updates P to Coyote `rc7` which introduces typed transitions for state machines: `RaiseEvent`, `GotoState`, `PushState` and `PopState` all now return a `Transition` (accordingly for a monitor). I am using the same approach that you do for monitors to handle these at the top level handlers and return back the transition. This is most likely the last major API change in Coyote as we are trying to polish and stabilize the APIs.

### CODE REVIEW

**Comment by ankushdesai** [comment]
The build is failing on a single test case. Let us also discuss these changes.

**Comment by pdeligia** [comment]
> The build is failing on a single test case. Let us also discuss these changes.

Everything passes now!

File: Src/Pc/CompilerCore/Backend/Coyote/CoyoteCodeGenerator.cs [hunk 1]
```diff
@@ -775,6 +757,11 @@ private void WriteStmt(CompilationContext context, StringWriter output, IPStmt s
                     break;
 
                 case FunCallStmt funCallStmt:
+                    if (funCallStmt.Function.CanChangeState == true || funCallStmt.Function.CanRaiseEvent == true)
```
Comment (ankushdesai) [inline Src/Pc/CompilerCore/Backend/Coyote/CoyoteCodeGenerator.cs]: @pdeligia I just realized this thing would not work out. Consider the function:
```
foo() {
if($) 
  goto S;
}
```
In other words, we will always early terminate on a function call that can change state or raise an event in its function body.

Comment (pdeligia) [reply Src/Pc/CompilerCore/Backend/Coyote/CoyoteCodeGenerator.cs]: Good catch! This should be now fixed.

**Comment by ankushdesai** [comment]
Thanks for these changes.

**Pull request merged**